index.js

```diff
@@ -39,6 +39,21 @@ export default class TabContainerElement extends HTMLElement {
       selectTab(this, index)
     })
   }
+
+  connectedCallback() {
+    for (const tab of this.querySelectorAll('[role="tablist"] [role="tab"]')) {
+      if (!tab.hasAttribute('aria-selected')) {
+        tab.setAttribute('aria-selected', 'false')
+      }
+      if (!tab.hasAttribute('tabindex')) {
+        if (tab.getAttribute('aria-selected') === 'true') {
+          tab.setAttribute('tabindex', '0')
+        } else {
+          tab.setAttribute('tabindex', '-1')
+        }
+      }
+    }
+  }
```

Comment:
This is a bit magical which I don't like.

I do like that we do the right thing even when the implementer doesn't follow the HTML instructions correctly. Since there is no immediate feedback to the implementer at runtime, they might half read the docs, get something working and go from there, not setting the correct `tabindex` and `aria-selected` attributes. It sucks if that work ends up resulting a degraded experience for screenreaders.

Alternatively we could add linting rules to the HTML to make sure the correct markup is applied.

Reply:
I had thought about this a bit more and I think it makes sense.

A lot of the ARIA attributes come with a promise that the component should behave a certain way, and those expectation is only fulfilled when the custom element gets upgraded. Therefore it makes sense that we only make that promise when JS comes into play.